internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -675,6 +676,37 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"key_vault_kms": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Required: true,
+						},
+						"key_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyVaultValidate.NestedItemId,
```

Comment:
Would `KeyID` here be better?

Reply:
keyVaultValidate.KeyID did nod work

Reply:
Yeah you're right, it seems like the `KeyID` function is no longer used.